webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -2480,19 +2480,135 @@ var CURRENT_GROUP_DATA={};
                     _this.toggleMediaOptions(this);
 
                 });
-                $mck_text_box.keyup(function (){
+                 _this.cursorPosition = function (element) {
+                    var selection = window.getSelection();
+                    var nodeCount = 0;
+                    var emptyNodes=0;
+                    var currentRange = selection.getRangeAt(element);
+                    var endOffset = currentRange.startOffset;
+                    var startNode = element.firstChild;
+                    var normalSpan = document.getElementById('mck-normal-span');
+                    var currentNode = selection.focusNode;
+                    if(normalSpan) {
+                        startNode = normalSpan.firstChild;
+                    }
+                    var newRange = document.createRange();
+                    newRange.setStart(startNode,0);
+                    newRange.setEnd(currentNode,endOffset);
+                    var allNodes = newRange.commonAncestorContainer.childNodes;
+                    for(nodeCount = 0;nodeCount<allNodes.length;nodeCount++)
+                    { 
+                        if(allNodes[nodeCount].nodeType && allNodes[nodeCount].nodeType == "3" && allNodes[nodeCount].nodeValue == "") {
+                            emptyNodes++;
+                          }
+                      if(allNodes[nodeCount] == newRange.endContainer && allNodes[nodeCount].nodeType == "3") {
```

Comment:
@hardikparashar whats the purpose of allNodes[nodeCount].nodeType in the 'if' condition above this 'if'. if that can be undefined, same error can come here too.